stock_picking_auto_create_package/models/stock_picking.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2023 Raumschmiede Gmbh
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_round
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _auto_create_delivery_package(self) -> None:
+        self.ensure_one()
+        if self.picking_type_id.automatic_package_creation_mode == "single":
+            self._auto_create_delivery_package_single()
+        elif self.picking_type_id.automatic_package_creation_mode == "packaging":
+            self._auto_create_delivery_package_per_smallest_packaging()
+
+    def _auto_create_delivery_package_per_smallest_packaging(self) -> None:
```

Comment:
```suggestion
    def _is_auto_create_delivery_package_possible_for_move_line(self, move_line):
        move_line.ensure_one()
        if move_line.state == "cancel":
            return
        if float_compare(move_line.qty_done, 0, precision_rounding=move_line.product_uom_id.rounding) <= 0:
            return
        if move_line.result_package_id:
            return
        return True
    
    def _auto_create_delivery_package_per_smallest_packaging(self) -> None:
```

Reply:
added state filter and extracted in a separate method